.github/workflows/CI-tests.yml

```
@@ -0,0 +1,100 @@
+name: CI tests
+
+on: [push]
+
+env:
+  poetry-version: 1.1.2
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Cache Poetry virtual env
+      uses: actions/cache@v2
+      env:
+          cache-name: cache-poetry-virtualenv
+      with:
+          path: .venv
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-python-${{ matrix.python-version }}-${{ hashFiles('pyproject.toml') }}
```

question: do you need `${{ env.cache-name }}` if that's the same everywhere? or is this with a plan to backport to other projects?

---

suggestion: use a hash of the lockfile instead.

```suggestion
          key: ${{ runner.os }}-build-${{ env.cache-name }}-python-${{ matrix.python-version }}-${{ hashFiles('pyproject.lock') }}
```

Same for other cases below.

---

Thanks. I've fixed these.